### ISSUE TITLE
Crash when destroying VM

### ISSUE THREAD

**Issue by anon8675309** [Telmate/terraform-provider-proxmox#406]
I have a backup script which runs on the destroy phase.  It took a bit over 3 hours for the backup script to run (and it did complete successfully) and then the provider crashed before deleting the VM.  I suspect that there's something that is getting cleaned up sooner than it should, but I don't know how golang's memory management works, so that's just my best guess since I've never seen this when the backups go quickly.

```
null_resource.provisioning: Still destroying... [id=3950432756488112771, 3h9m11s elapsed]
null_resource.provisioning: Destruction complete after 3h9m21s
module.standard_vm.module.restore_backups.null_resource.install_backup_script: Destroying... [id=8065992563811535162]
module.standard_vm.module.updates.null_resource.updates: Destroying... [id=848424791315036291]
module.standard_vm.module.restore_backups.null_resource.install_backup_script: Destruction complete after 0s
module.standard_vm.module.updates.null_resource.updates: Destruction complete after 0s
module.standard_vm.module.copy_backups.null_resource.copy_backups: Destroying... [id=3759902521173098930]
module.standard_vm.module.copy_backups.null_resource.copy_backups: Destruction complete after 0s
module.standard_vm.module.vm.proxmox_vm_qemu.server: Destroying... [id=tc/qemu/111]
module.standard_vm.module.vm.proxmox_vm_qemu.server: Still destroying... [id=tc/qemu/111, 10s elapsed]
module.standard_vm.module.vm.proxmox_vm_qemu.server: Still destroying... [id=tc/qemu/111, 20s elapsed]
��
� Error: Plugin did not respond
�
� The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more
� details.
��

Stack trace from the terraform-provider-proxmox_v2.7.1 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 35 [running]:
github.com/Telmate/proxmox-api-go/proxmox.(*Client).GetVmInfo(0xc000297860, 0xc00011b6e8, 0x0, 0x0, 0x0)
        github.com/Telmate/proxmox-api-go@v0.0.0-20210507143528-c60bbda13c0c/proxmox/client.go:149 +0x645
github.com/Telmate/proxmox-api-go/proxmox.(*Client).CheckVmRef(...)
        github.com/Telmate/proxmox-api-go@v0.0.0-20210507143528-c60bbda13c0c/proxmox/client.go:142
github.com/Telmate/proxmox-api-go/proxmox.(*Client).StatusChangeVm(0xc000297860, 0xc00011b6e8, 0xca5780, 0x4, 0x0, 0x0, 0x203000, 0x203000)
        github.com/Telmate/proxmox-api-go@v0.0.0-20210507143528-c60bbda13c0c/proxmox/client.go:424 +0x5c
github.com/Telmate/proxmox-api-go/proxmox.(*Client).StopVm(...)
        github.com/Telmate/proxmox-api-go@v0.0.0-20210507143528-c60bbda13c0c/proxmox/client.go:448
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceVmQemuDelete(0xc0000e7100, 0xb75b40, 0xc000297d10, 0x0, 0x0)
        github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:1374 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).delete(0xc0000f49a0, 0xd9e098, 0xc00017cb40, 0xc0000e7100, 0xb75b40, 0xc000297d10, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:369 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0000f49a0, 0xd9e098, 0xc00017cb40, 0xc0002bc4d0, 0xc0002b9b40, 0xb75b40, 0xc000297d10, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:428 +0x8f7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000d3f8, 0xd9e098, 0xc00017cb40, 0xc000296d20, 0xc00017cb40, 0xc6f160, 0xc00019fb00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000162700, 0xd9e140, 0xc00017cb40, 0xc0002bc310, 0xc000162700, 0xc00019fb30, 0xc000199ba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xc6f160, 0xc000162700, 0xd9e140, 0xc00019fb30, 0xc0002ea420, 0x0, 0xd9e140, 0xc00019fb30, 0xc000273800, 0xaf9)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001bce00, 0xda57f8, 0xc000184d80, 0xc0000fa400, 0xc0001fd230, 0x11b0ce0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0001bce00, 0xda57f8, 0xc000184d80, 0xc0000fa400, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0002681b0, 0xc0001bce00, 0xda57f8, 0xc000184d80, 0xc0000fa400)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-proxmox_v2.7.1 plugin crashed!
```

The "interface {} is nil" error does appears in #347, but I think that's a separate bug.  Issue #193 looks similar, but doesn't have any stacktrace, so I am not sure if it's the same bug or not.

Here's my version info:
```
Terraform v0.15.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/telmate/proxmox v2.7.1

Proxmox version 7.0-11
```

**Comment by anon8675309** [comment]
I just ran into this using `terraform-provider-proxmox_v2.9.8`.

I tried to create a minimal test case which will reliably reproduce this issue, but sadly I get a different error (401 - Permission denied).  I've created #561 to track that, and I am hoping that it might lead to an answer about this "interface {} is nil, not []interface {}" error.

**Comment by anon8675309** [comment]
I just retested with v2.9.10 of the provided and confirmed that this no longer crashes.  As the test case from #561 was previously reliably triggering this problem, I must conclude that this issue has been fixed.

This issue was similar to #347 and #466, so if those can be re-tested with v2.9.10, it may be possible to close those as well.

For postarity, here's the version info for my latest test:  
```
Terraform v1.2.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/telmate/proxmox v2.9.10
```
